### PR TITLE
consul/connect: enable setting local_bind_address in upstream

### PR DESCRIPTION
**Pull request by Ilhicas**
Upstreams configuration in service->proxy->upstream stanza currently doesn't allow for the upstream local_bind_address definition

This allows to define different local_bind_addresses allowing envoy to read values from consul xds for example to bind the upstream to 127.0.0.X or others, allowing it to effectively avoid port binding in upstreams.

As an example

127.0.0.1:5432 
127.0.0.2:5432

Could be used in multiple upstream stanzas without port binding issues.

This field is optional and defaults to the local_bind_address.

Closes https://github.com/hashicorp/nomad/issues/6248

More info on the motivations can be found at : https://github.com/hashicorp/nomad/issues/10057

### CODE REVIEW

**Comment by jorgensoares** [comment]
@tgross can we have this reviewed :) thank you!

**Comment by tgross** [comment]
@jorgensoares we're in the middle of prepping the Nomad 1.0.4 release but it'll get reviewed soon.

**Comment by Ilhicas** [comment]
Hi regarding the changelog I added it there, but might not make sense, since I noticed other PR's were adding it 1.0.4 as well.
So might have to set it elsewhere?
What is approriate here?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,5 @@
 ## 1.0.4 (Unreleased)
+ * consul/connect: Enable setting `local_bind_address` field on connect upstreams [[GH-6248](https://github.com/hashicorp/nomad/issues/6248)]
```
Comment (shoenig) [inline CHANGELOG.md:2]: Like you already mentioned, we'll want this under the `1.1.0` header now, specifically under the `IMPROVEMENTS` sub header

**Review by shoenig — CHANGES_REQUESTED** [review]
This is looking really great @Ilhicas, thanks for the PR!

There's a block of code where we interpolate fields, and we'll want to add support for interpolating the local_bind_address also

https://github.com/hashicorp/nomad/blob/d1648243f45cccdd900c93f2cf1c2b7f643536d5/client/taskenv/services.go#L131

Just that and the CHANGELOG shuffle, I think this will be good to go.

File: CHANGELOG.md (hunk 1 above)
Comment (Ilhicas) [reply CHANGELOG.md:9]: Updated changelog

Also added LocalBindAddress to interpolation function, and updated interpolation tests

**Review by shoenig — APPROVED** [review]
LGTM; thanks @Ilhicas !

**Pull request merged**